tests/conftest.py

```diff
@@ -270,177 +211,86 @@ def dummy_carbon_data(fixture_core_components):
+    ] = [-3.0, -10.0, -250.0, -10000.0]
+
+    data["soil_temperature"] = fixture_core_components.layer_structure.from_template()
+    data["soil_temperature"].loc[
+        {"layers": fixture_core_components.layer_structure.role_indices["topsoil"]}
+    ] = [35.0, 37.5, 40.0, 25.0]
+    data["soil_temperature"].loc[
+        {"layers": fixture_core_components.layer_structure.role_indices["subsoil"]}
+    ] = [22.5, 22.5, 22.5, 22.5]
 
     return data
 
 
 @pytest.fixture
 def top_soil_layer_index(fixture_core_components):
-    """The index of the top soil layer in the data fixtures."""
-    return fixture_core_components.layer_structure.layer_roles.index("soil")
-
+    """The index of the top soil layer in the data fixtures.
 
-@pytest.fixture
-def surface_layer_index(fixture_core_components):
-    """The index of the top soil layer in the data fixtures."""
-    return fixture_core_components.layer_structure.layer_roles.index("surface")
+    Convert from array to scalar using item.
+    """
+    return fixture_core_components.layer_structure.role_indices["topsoil"].item()
 
 
 @pytest.fixture
-def new_axis_validators():
-    """Create new axis validators to test methods and registration."""
-    from virtual_ecosystem.core.axes import AxisValidator
-    from virtual_ecosystem.core.grid import Grid
-
-    # Create a new subclass.
-    class TestAxis(AxisValidator):
-        core_axis = "testing"
-        dim_names = frozenset(["test"])
-
-        def can_validate(self, value: DataArray, grid: Grid, **kwargs: Any) -> bool:
-            return True if value.sum() > 10 else False
-
-        def run_validation(
-            self, value: DataArray, grid: Grid, **kwargs: Any
-        ) -> DataArray:
-            return value * 2
-
-    # Create a new duplicate subclass to check mutual exclusivity test
-    class TestAxis2(AxisValidator):
-        core_axis = "testing"
-        dim_names = frozenset(["test"])
-
-        def can_validate(self, value: DataArray, grid: Grid, **kwargs: Any) -> bool:
-            return True if value.sum() > 10 else False
+def surface_layer_index(fixture_core_components):
+    """The index of the top soil layer in the data fixtures.
 
-        def run_validation(
-            self, value: DataArray, grid: Grid, **kwargs: Any
-        ) -> DataArray:
-            return value * 2
+    Convert from array to scalar using item.
+    """
+    return fixture_core_components.layer_structure.role_indices["surface"].item()
 
 
 @pytest.fixture
 def dummy_climate_data(fixture_core_components):
     """Creates a dummy climate data object for use in tests."""
 
     from virtual_ecosystem.core.data import Data
-    from virtual_ecosystem.core.grid import Grid
 
     # Setup the data object with four cells.
-    grid = Grid(
-        grid_type="square",
-        cell_nx=3,
-        cell_ny=1,
-        cell_area=3,
-        xoff=0,
-        yoff=0,
-    )
-    data = Data(grid)
+    data = Data(fixture_core_components.grid)
```

Comment:
Just to double-check, did you change the number of cell_ids for all my tests from 3 to 4?

Reply:
I did. 

I'm not happy about it and I'm still trying to decide if it is the right thing to do. 

Currently we have models that use 4 cells and models that use 3. The underlying issue is that we have a big pile of fixtures that are used more or less as a pick and mix stall. That's sorta ok, but we end up hand coding indices all over the place (3 v. 4 for example) to make the selection work together. With the more integrated core components, that's going to lead to more inconsistencies in tests and stuff basically not working because half the test uses one setup and  the other half uses something else.

Ultimately, if everything uses `fixture_core_components` and a small number of extra data frames, then updating tests etc. becomes much cleaner. 

It's a massive pain in the arse: I _think_ it is the right thing to do, but this PR went down a rabbit hole on Friday and I just need to figure out what to do.

Reply:
I see and I understand. It's become a bit of a mess with the fixtures and conftest file in general.
I totally agree that this is something that needs sorting, but in the interest of keeping PRs manageable I suggest to focus on what we discussed, add more features to the layer structure and get rid of the subcanopy layer, make that work, merge in develop and branch off again to do the next step, like a PR that just focuses on tidying up the test fixtures. Would you agree?

Reply:
This also links to the overall comment I made about developing other features on parallel branches, which is becoming an absolute nightmare if we make too many changes here at once.

I will pause the stats axis for now #437 (it's not required for min viable product), it just makes it more confusing with the layer structure and the variables that Diego is working on. Once these things are in place it should be much easier to add an axis and update the relevant locations.

Reply:
> I suggest to focus on what we discussed, add more features to the layer structure and get rid of the subcanopy layer, make that work, merge in develop and branch off again to do the next step

I think that's probably right - let me run with this for today - I've got a bunch of fixes to get the tests working that I'll probably put into a sub PR.

Reply:
> get rid of the subcanopy layer

Just to note - this is the tricky bit. The tests all need updating to remove this, so cleaning them to make this process cleaner is kinda built in.

Reply:
I agree, the same thing happened to me with the climate_stats, which is why I think it makes more sense for me to  be here with you and respond where needed or contribute rather than making different changes on a different branch.